src/solvers/adapters.jl

```diff
@@ -0,0 +1,32 @@
+
+#
+# TupleRegionIterator
+#
+# Adapts outputs to be (region, region_kwargs) tuples
+#
+# More generic design? maybe just assuming RegionIterator
+# or its outputs implement some interface function that
+# generates each tuple?
+#
+
+mutable struct TupleRegionIterator{RegionIter}
```

Comment:
Maybe we can come up with a name that is more descriptive, like `RegionIteratorReturnKwargs`. `Tuple` is a bit vague.